[ Jekyll ] [ SpecInsert ] omit_header (#8766)
opensearch-project/documentation-website: DEVELOPER_GUIDE.md

<!-- spec_insert_start 
api: <API_NAME>
component: <COMPONENT_NAME>

> other_argument: <OTHER_ARGUMENT>

Author: Calling this argument to match what the code is actually using and avoid confusion with the API path and query params.

bundle exec jekyll spec-insert --refresh-spec
```

### Ignoring files and folders
The `spec-insert` plugin ignores all files and folders listed in the [./_config.yml#exclude](./_config.yml) list, which is also the list of files and folders that Jekyll ignores.

## CI/CD
The `spec-insert` plugin is run as part of the CI/CD pipeline to ensure that the API components are up to date in the documentation. This is performed through the [update-api-components.yml](.github/workflows/update-api-components.yml) GitHub Actions workflow, which creates a pull request containing the updated API components every Sunday.

## Spec insert components
All spec insert components accept the following arguments:
- `api` (String; required): The name of the API to render the component from. This is equivalent to the `x-operation-group` field in the OpenSearch OpenAPI Spec.

> - `component` (String; required): The name of the component to render (e.g., `query_parameters`, `path_parameters`, or `paths_and_http_methods`).

Reviewer: ```suggestion
- `component` (String; required): The name of the component to render,  such as `query_parameters`, `path_parameters`, or `paths_and_http_methods`.
```

> - `omit_header` (Boolean; default is `false`): If set to `true`, the markdown header of the component (e.g. `## Query parameters`) will not be rendered.

Reviewer: ```suggestion
- `omit_header` (Boolean; Default is `false`): If set to `true`, the markdown header of the component will not be rendered.
```
